cosmo/serializer.py

```diff
@@ -288,6 +289,56 @@ def _get_unit(self, iface):
 
             self.l3vpns[iface["vrf"]["id"]]["interfaces"].append(iface["name"])
 
+        if tags.has("cpe", "bgp"):
+            groupname = "CPE_" + iface["name"].replace(".", "-").replace("/", "-")
+
+            if iface["vrf"]:
+                vrfname = [v["name"] for v in self.vrfs if v["id"] == iface["vrf"]["id"]][0]
+            else:
+                vrfname = "default"
+
+            # ensure paths exist
+            if not vrfname in self.routing_instances:
+                self.routing_instances[vrfname] = {}
+            if not "protocols" in self.routing_instances[vrfname]:
+                self.routing_instances[vrfname] = { "protocols": { "bgp": {"groups": {} } } }
+
+            policy_v4 = { "import_list": [] }
+            policy_v6 = { "import_list": [] }
+            if vrfname == "default":
+                policy_v4["export"] = "DEFAULT_V4"
+                policy_v6["export"] = "DEFAULT_V6"
+
+            if iface["parent"]:
+                parent = [i for i in self.device["interfaces"] if i["id"] == iface["parent"]["id"]][0]
+                connected = parent["connected_endpoints"][0]["device"]
+                addresses = set()
+                for a in connected["interfaces"]:
+                    for i in a["ip_addresses"]:
+                        if i["address"] == connected["primary_ip4"]["address"]:
```

Comment:
What does this do? It excludes the primary IP from the device? Is this meant to exclude management networks?